packages/dev/core/src/FlowGraph/Blocks/Execution/flowGraphTimerBlock.ts

```diff
@@ -0,0 +1,67 @@
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+import type { FlowGraphSignalConnection } from "../../flowGraphSignalConnection";
+import { AdvancedTimer } from "../../../Misc/timer";
+import type { FlowGraphContext } from "../../flowGraphContext";
+import { FlowGraphAsyncExecutionBlock } from "../../flowGraphAsyncExecutionBlock";
+import { RichTypeNumber } from "../../flowGraphRichTypes";
+
+/**
+ * @experimental
+ * Block that can execute an action immediately and another after a delay.
+ * The delay is counted on the scene's tick.
+ */
+export class FlowGraphTimerBlock extends FlowGraphAsyncExecutionBlock {
+    public readonly timeout: FlowGraphDataConnection<number>;
+    public readonly onTimerDone: FlowGraphSignalConnection;
+
+    constructor() {
+        super();
+
+        this.timeout = this._registerDataInput("timeout", RichTypeNumber);
+        this.onTimerDone = this._registerSignalOutput("onTimerDone");
+    }
+
+    public _preparePendingTasks(context: FlowGraphContext): void {

```

Comment:
is the only way to configure the timeout is using the context? No way to configure during construction?

Reply:
I can add an optional constructor parameter to configure the timeout

Reply:
i think it would make sense to add init options to the constructor(s)